src/yarpmotorgui/main.cpp

```diff
@@ -158,6 +157,11 @@ int main(int argc, char *argv[])
                     }
                 }
             }
+
+            if (desc_driver)
+            {
+                delete desc_driver;
+            }
```

Comment:
I'm not sure about what is happening here, can you explain why you are deleting the polydriver? Is it no longer needed?

Reply:
You are right, I missed it I thought the main terminated there.
BTW before deleting the pointer it I would call `desc_driver->close()`